lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.h

```diff
@@ -848,10 +855,22 @@ class ProxyShape
   AL_USDMAYA_PUBLIC
   SdfPathVector getPrimPathsFromCommaJoinedString(const MString &paths) const;
 
+#if defined(WANT_UFE_BUILD)
+  /// \brief Get the UFE path of the maya proxy shape
+  /// \return An UFE path containing the path to the proxy shape
+  AL_USDMAYA_PUBLIC
+  Ufe::Path ufePath() const;
+#endif
+
   /// \brief  Returns the selection mask of the shape
   AL_USDMAYA_PUBLIC
   MSelectionMask getShapeSelectionMask() const override;
 
+  /// \brief  Clears the bounding box cache of the shape
+  AL_USDMAYA_PUBLIC
```

Comment:
Probably a pointless use of DSO exporting here. inline would suffice for such a nothing method :)

Reply:
I wasn't the dev who originally coded this, but why wouldn’t the clear bounding box cache be useful outside the shared library?

Reply:
There are two types of function available to exe's/dso's outside of this lib.
1. inline methods.
2. DSO exported methods. 
In this case we have a single line method that simply calls another method - aka, a tail call. DSO exporting a tail call is kinda nuts _(since it inserts extra symbols in to the DSO symbol table, adds a bit of overhead on startup, incurs a small penalty when you factor in the additional jumps needed to call the wrapped method, and also impedes branch prediction and HW prefetch)_

I've tried to put together a tiny code sample to explain my point:
https://godbolt.org/z/GjB-hP

In this case, If clearBoundingBoxCache is DSO exported, we have a tail call chain like so:

```someFuncInMyExe() -->  clearBoundingBoxCache()  -->  std::map<...>::clear()```

Given that clearBoundingBoxCache() doesn't actually do anything useful, if you inline it, the compiler removes the pointless jump and calls clear directly. 

```someFuncInMyExe() -->  std::map<...>::clear()```

Given that the UsdTimeCode and MBoundingBox are already DSO exported types, and let's assume that std::map is actually a standardised template (*ahem*), then any exe making use of the library has enough information to be able to clear the cache. 

Sorry, this is just one of my major code gripes. It's not the biggest crime in the world, that would be this:

```c++
class Vec3
{
public:
   DLL_EXPORT float getX() const;
   DLL_EXPORT float getY() const;
   DLL_EXPORT float getZ() const;
}:
```
Which is _**literally**_ infinity percent slower than inline :(

Reply:
Thanks for the detailed explanation. We agree and have changed it.